kong/core/globalpatches.lua

```diff
@@ -49,6 +49,27 @@ return function(options)
 
 
 
+  do -- implement `sleep` in the `init_worker` context
+
+    -- initialization code regularly uses the shm and locks.
+    -- the resty-lock is based on sleeping while waiting, but that api
+    -- is unavailable. Hence we implement a BLOCKING sleep, only in
+    -- the init_worker context.
+    local get_phase= ngx.get_phase
+    local ngx_sleep = ngx.sleep
+    local alternative_sleep = require("socket").sleep
+    ngx.sleep = function(s)
+      if get_phase() == "init_worker" then
+        ngx.log(ngx.WARN, "executing a blocking 'sleep' (", s, " seconds)")
+        return alternative_sleep(s)
+      end
+      return ngx_sleep(s)
+    end
+
+  end
```

Comment:
I thought about this, but it is not a `lua-cassandra` problem, it's an omission in OpenResty. I've had the exact same issue with `resty-worker-events`.

In the `init_worker` phase multiple workers can start simultaneously, and hence require synchronization. The `resty-lock` is based on `shm` storage and the `sleep` method. 
So I think this is the better solution. Fix it in `lua-cassandra` and it will bite us again with some other module.

In the `init_worker` phase we also do not really care about the blocking nature (`resty-lock` uses 1ms intervals). For safety I added the log message so we can track it if it goes sour.

If anything, we should create an upstream openresty patch.